interfaces/builtin/posix_mq.go

```diff
@@ -0,0 +1,237 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/snap"
+)
+
+const posixMQSummary = `allows access to POSIX message queues`
+
+const posixMQBaseDeclarationSlots = `
+  posix-mq:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-auto-connection: false
```

Comment:
We probably want to be more restrictive on this, otherwise a snap could be able to impersonate another snap and read/write messages which were not meant for it.

Using a declaration similar to that of the shared-memory interface might make sense here too.